test/gtest/db_sync.cpp

```diff
@@ -422,6 +422,26 @@ bool CheckKDBForTargetID(const fs::path& filename)
     }
     return count != 0;
 }
+
+bool CheckKDBJournalMode(const fs::path& filename)
+{
+    auto journal_query = "PRAGMA journal_mode";
+    auto sql           = SQLite{filename.string(), true};
+    auto stmt          = SQLite::Statement{sql, journal_query};
+    std::string journal_mode;
+    while(true)
+    {
+        auto rc = stmt.Step(sql);
+        if(rc == SQLITE_ROW)
+            journal_mode = stmt.ColumnText(0);
+        else if(rc == SQLITE_DONE)
+            break;
+        else if(rc == SQLITE_ERROR || rc == SQLITE_MISUSE)
+            throw std::runtime_error(sql.ErrorMessage());
```

Comment:
will it go into infinite loop if `sqlite3_step` return `SQLITE_BUSY` or `SQLITE_LOCKED`?

Reply:
@cderb @bghimireamd I would recommend to be safe and just replace this with `else`. What do you think? CC: @JehandadKhan